subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/MavenConversionDynamicPomIntegrationTest.groovy

```diff
@@ -0,0 +1,143 @@
+        targetDir.file("pom.xml") << """
+            <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+                <modelVersion>4.0.0</modelVersion>
+
+                <groupId>util</groupId>
+                <artifactId>util</artifactId>
+                <version>2.5</version>
+                <packaging>jar</packaging>
+
+                <dependencies>
+                    <dependency>
+                        <groupId>commons-lang</groupId>
+                        <artifactId>commons-lang</artifactId>
+                        <version>2.6</version>
+                    </dependency>
+                    <dependency>
+                        <groupId>junit</groupId>
+                        <artifactId>junit</artifactId>
+                        <version>4.13.1</version>
+                        <scope>test</scope>
+                    </dependency>
+                </dependencies>
+                <build>
+                    <plugins>
+                        <plugin>
+                            <groupId>org.apache.maven.plugins</groupId>
+                            <artifactId>maven-compiler-plugin</artifactId>
+                            <configuration>
+                                <source>${source}</source>
+                                <target>${target}</target>
+                            </configuration>
+                        </plugin>
+                    </plugins>
+                </build>
+            </project>
+        """
+        targetDir.file("src/main/java/Foo.java") << """
+            import org.apache.commons.lang.StringUtils;
+
+            public class Foo {
+              public String toString() {
+                return StringUtils.normalizeSpace("hi  there!");
+              }
+            }
+        """
+        targetDir.file("src/test/java/FooTest.java") << """
+            import org.junit.Test;
+
+            public class FooTest {
+              @Test public void test() {
+                assert false: "test failure";
+              }
+            }
+        """
+
+        def dsl = dslFixtureFor(scriptDsl)
+
+        when:
+        run 'init', '--dsl', scriptDsl.id as String
+
+        then:
+        dsl.assertGradleFilesGenerated()
+        dsl.getSettingsFile().text.contains("rootProject.name = 'util'") || dsl.getSettingsFile().text.contains('rootProject.name = "util"')
+        MavenConversionIntegrationTest.assertContainsPublishingConfig(dsl.getBuildFile(), scriptDsl)
+        dsl.getBuildFile(targetDir).text.contains("java.sourceCompatibility = JavaVersion.${source.name()}")
+        // only shows up if the target is different
+        def targetDiffers = source != target
+        targetDiffers == dsl.getBuildFile(targetDir).text.contains("java.targetCompatibility = JavaVersion.${target.name()}")
```

Comment:
❌ I really dislike conditional assertions like this.  Can we just split this into two tests?

- Move the source code up into a setup()
- Add a `writePom(source, target)` method to write the POM
- In one test, have source == target and in the other source != target. 

❓ do we need to test a case where only one of source or target are set?

Reply:
Yeah, I had it like that, but I didn't like the repetitiveness. I'll try to adjust it.